web/helpers/authentication_helpers.ex

```diff
@@ -28,16 +28,21 @@ defmodule CodeCorps.AuthenticationHelpers do
   # This is partially adjusted code, taken from canary
   def load_and_authorize_changeset(conn, options) do
     action = conn.private.phoenix_action
-
     cond do
-      action in options[:only] -> conn |> load_resource(options) |> do_init_and_authorize_changeset(options[:model], action)
-      true -> conn
+      action in options[:only] ->
+        conn
+        |> load_resource(options)
+        |> do_init_and_authorize_changeset(options[:model], action)
+      true ->
+        conn
     end
   end
 
   defp do_init_and_authorize_changeset(conn, model, action) do
     changeset = init_changeset(conn, model, action)
-    conn |> assign(:changeset, changeset) |> authorize(changeset, action)
+    conn
+    |> assign(:changeset, changeset)
+    |> authorize(changeset, action)
```

Comment:
This works really funky with `Preview` because we're doing an `assign_user` inside of the changeset. When this plug runs before the controller code itself is executed, we don't have the user that's going to be assigned in the controller yet inside the changeset that checks whether we're going to authorize it.

This is a bit 🐔 and 🍳 as a result.